video/sweeper.go

```diff
@@ -0,0 +1,71 @@
+package video
+
+import (
+	"sort"
+	"sync"
+	"sync/atomic"
+)
+
+// A Counter is a thread-safe counter implementation
+type counter uint64
+
+// Incr method increments the counter by some value
+func (c *counter) Inc() {
+	atomic.AddUint64((*uint64)(c), 1)
+}
+
+// Value method returns the counter's current value
+func (c *counter) Value() uint64 {
+	return atomic.LoadUint64((*uint64)(c))
+}
+
+type sweeper struct {
+	counters sync.Map
```

Comment:
you sure `sync.Map` is what you want here? just checking because it's not meant to be a generic "concurrent map". the docs for sync.Map describe the limited set of cases where it should be used. outside those, they recommend implementing your own concurrent map.

not saying you're wrong. just a mistake ive seen in other places.

Reply:
Simplified with a mutex and it turned out to be faster.